apps/meteor/client/views/admin/ABAC/ABACAttributeField.tsx

```diff
@@ -0,0 +1,102 @@
+import { Box, Button, FieldError, FieldRow, InputBoxSkeleton, MultiSelect, PaginatedSelectFiltered } from '@rocket.chat/fuselage';
+import { useDebouncedValue } from '@rocket.chat/fuselage-hooks';
+import { useCallback, useMemo, useState } from 'react';
+import { Controller, useFormContext, useWatch } from 'react-hook-form';
+import { useTranslation } from 'react-i18next';
+
+import type { AdminABACRoomFormData } from './AdminABACRoomForm';
+import useABACAttributeList from './hooks/useABACAttributeList';
+
+type ABACAttributeAutocompleteProps = {
+	onRemove: () => void;
+	index: number;
+};
+
+const ABACAttributeField = ({ onRemove, index }: ABACAttributeAutocompleteProps) => {
+	const {
+		formState: { errors },
+		control,
+		getValues,
+		setValue,
+	} = useFormContext<AdminABACRoomFormData>();
+	const currentAttribute = useWatch({ control, name: `attributes.${index}.key` });
+	const { t } = useTranslation();
+	const [filter, setFilter] = useState<string>();
+	const filterDebounced = useDebouncedValue(filter, 300);
+
+	const { data: options, fetchNextPage, isLoading } = useABACAttributeList(filterDebounced || undefined);
+
+	const selectedAttribute = options.find((option) => option.value === currentAttribute);
+
+	const validateRepeatedAttributes = useCallback(
+		(value: string) => {
+			const attributes = getValues('attributes');
+			// Only one instance of the same attribute is allowed to be in the form at a time
+			const repeatedAttributes = attributes.filter((attribute) => attribute.key === value && attribute.key !== currentAttribute).length > 1;
+			return repeatedAttributes ? t('ABAC_No_repeated_attributes') : undefined;
+		},
+		[currentAttribute, getValues, t],
+	);
+
+	const valueOptions: [string, string][] = useMemo(() => {
+		if (!selectedAttribute?.attributeValues) {
+			return [];
+		}
+		return selectedAttribute.attributeValues.map((value) => [value, value]);
+	}, [selectedAttribute]);
+
+	if (isLoading) {
+		return <InputBoxSkeleton />;
+	}
+	return (
+		<Box display='flex' flexDirection='column' w='full'>
+			<FieldRow>
+				<Controller
```

Comment:
Might be a good idea to consider using `useController` instead of `<Controller />` here. It will provide you with a `fieldState` value, which contains only the information relevant to this field, like error, invalid, etc. It would reduce the number of places where you need to do things like `errors.attributes.${index}.${fieldName}`.

Docs: https://react-hook-form.com/docs/usecontroller
Example in our codebase: [ChannelField.tsx](https://github.com/RocketChat/Rocket.Chat/blob/develop/apps/meteor/client/views/omnichannel/components/outboundMessage/components/OutboundMessageWizard/forms/RecipientForm/components/ChannelField.tsx)